Polyfills/Canvas/Source/Image.cpp

```diff
@@ -65,13 +79,42 @@ namespace Babylon::Polyfills::Internal
         return Napi::Value::From(Env(), m_src);
     }
 
-    void NativeCanvasImage::SetSrc(const Napi::CallbackInfo&, const Napi::Value& value)
+    void NativeCanvasImage::SetSrc(const Napi::CallbackInfo& info, const Napi::Value& value)
     {
         auto text{value.As<Napi::String>().Utf8Value()};
-        m_src = text;
+        UrlLib::UrlRequest request{};
+        request.Open(UrlLib::UrlMethod::Get, text);
+        request.ResponseType(UrlLib::UrlResponseType::Buffer);
+        request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{ info.Env() }, this, request{std::move(request)}](arcana::expected<void, std::exception_ptr> result) {
```

Comment:
I think you must use a cancellationToken here. If the runtime goes down before this request completes, the continuation code will probably not be happy.